### ISSUE TITLE
Hardcoded dependency on glibc causes a crash

### ISSUE THREAD

**Issue by Frontear** [ChloePrime/AAAParticles#35]
Relevant log entries (this is from a LPS modpack called Prominence, I can share a similar crash report with just AAAParticles if necessary)

```
java.lang.RuntimeException: Could not execute entrypoint stage 'client' due to errors, provided by 'aaa_particles'!
    at app//net.fabricmc.loader.impl.FabricLoaderImpl.lambda$invokeEntrypoints$2(FabricLoaderImpl.java:388)
    at app//net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
    at app//net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:386)
    at app//net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:53)
    at MC//net.minecraft.client.MinecraftClient.<init>(MinecraftClient.java:458)
    at net.minecraft.client.main.Main.main(Main.java:211)
    at app//net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
    at app//net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
    at app//net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
    at app//org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:100)
    at app//org.prismlauncher.EntryPoint.listen(EntryPoint.java:129)
    at app//org.prismlauncher.EntryPoint.main(EntryPoint.java:70)
Caused by: java.lang.ExceptionInInitializerError
    at mod.chloeprime.aaaparticles.client.AAAParticlesClient.installNativeLibrary(AAAParticlesClient.java:55)
    at mod.chloeprime.aaaparticles.client.AAAParticlesClient.init(AAAParticlesClient.java:25)
    at mod.chloeprime.aaaparticles.fabric.AAAParticlesFabricClient.onInitializeClient(AAAParticlesFabricClient.java:15)
    at app//net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:384)
    ... 9 more
Caused by: java.lang.UnsatisfiedLinkError: /home/ewerd/.local/share/PrismLauncher/instances/Prominence II RPG- Hasturian Era/minecraft/libEffekseerNativeForJava.so: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.38' not found (required by /home/ewerd/.local/share/PrismLauncher/instances/Prominence II RPG- Hasturian Era/minecraft/libEffekseerNativeForJava.so)
    at java.base@17.0.12/jdk.internal.loader.NativeLibraries.load(Native Method)
    at java.base@17.0.12/jdk.internal.loader.NativeLibraries$NativeLibraryImpl.open(NativeLibraries.java:388)
    at java.base@17.0.12/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:232)
    at java.base@17.0.12/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:174)
    at java.base@17.0.12/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2394)
    at java.base@17.0.12/java.lang.Runtime.load0(Runtime.java:755)
    at java.base@17.0.12/java.lang.System.load(System.java:1957)
    at mod.chloeprime.aaaparticles.client.AAAParticlesClient.installNativeLibrary(AAAParticlesClient.java:53)
    ... 12 more
```

**Comment by Frontear** [comment]
Honestly this issue is probably realistically unfeasible to solve, given it occurred on a Debian 12 Stable, a distro known to be behind on packages. Wanted to report it anyways though

**Comment by ChloePrime** [comment]
This happens on 1.4.x or 1.3.1?

**Comment by Frontear** [comment]
1.4.3 according to the user who experienced this.

**Comment by ripjackie** [comment]
This bug is a lot harder to pin down and fix, due to this being an issue building on a distro like arch. Some symbols in effekseer/Effekseer are building with the highest possible GLIBC version, and for the life of me I cannot get them to behave. I'll keep working on this.

**Comment by ripjackie** [comment]
Hopefully a fix possible as mentioned in #36. Asking users to compile their own libs is a pretty big ask. I'll look into compiling on an earlier glibc version or forgoing glibc specific functions entirely. lots to do with that statement, though.

**Comment by Hello-User** [comment]
I'm on fully updated Fedora 40 KDE. Unfortunately, compiling #36 results in the same error. Using AAAParticles version 1.4.3.

**Comment by Frontear** [comment]
Crash in regards to GLIBC?

**Comment by ripjackie** [comment]
@Hello-User which error are you referring to?
- if the glibc error, please show the output of `ldd --version`
- if the libglu error, please show the output of `ldd libEffekseerNativeForJava.so`

**Comment by Hello-User** [comment]
> @Hello-User which error are you referring to?
> 
>     * if the glibc error, please show the output of `ldd --version`
> 
>     * if the libglu error, please show the output of `ldd libEffekseerNativeForJava.so`

Apologies, I'm referring to the libGLU error. Interestingly, the self-compiled library gets replaced when the game starts.

Before starting the game:
```
ldd libEffekseerNativeForJava.so
        linux-vdso.so.1 (0x00007f1d038f8000)
        libGLX.so.0 => /lib64/libGLX.so.0 (0x00007f1d038a4000)
        libOpenGL.so.0 => /lib64/libOpenGL.so.0 (0x00007f1d0387f000)
        libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007f1d03200000)
        libm.so.6 => /lib64/libm.so.6 (0x00007f1d0351c000)
        libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f1d034ee000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f1d0300f000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f1d038fa000)
        libX11.so.6 => /lib64/libX11.so.6 (0x00007f1d02eca000)
        libXext.so.6 => /lib64/libXext.so.6 (0x00007f1d034da000)
        libGLdispatch.so.0 => /lib64/libGLdispatch.so.0 (0x00007f1d02e51000)
        libxcb.so.1 => /lib64/libxcb.so.1 (0x00007f1d034af000)
        libXau.so.6 => /lib64/libXau.so.6 (0x00007f1d034a9000)
```

After starting:
```
ldd libEffekseerNativeForJava.so
ldd: warning: you do not have execution permission for `./libEffekseerNativeForJava.so'
        linux-vdso.so.1 (0x00007fd093d02000)
        libOpenGL.so.0 => /lib64/libOpenGL.so.0 (0x00007fd093cba000)
        libGLX.so.0 => /lib64/libGLX.so.0 (0x00007fd093c89000)
        libGLU.so.1 => /lib64/libGLU.so.1 (0x00007fd093c32000)
        libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007fd093200000)
        libm.so.6 => /lib64/libm.so.6 (0x00007fd09351c000)
        libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007fd093c02000)
        libc.so.6 => /lib64/libc.so.6 (0x00007fd09300f000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fd093d04000)
        libGLdispatch.so.0 => /lib64/libGLdispatch.so.0 (0x00007fd0934a3000)
        libX11.so.6 => /lib64/libX11.so.6 (0x00007fd092eca000)
        libXext.so.6 => /lib64/libXext.so.6 (0x00007fd093bee000)
        libxcb.so.1 => /lib64/libxcb.so.1 (0x00007fd093bc1000)
        libXau.so.6 => /lib64/libXau.so.6 (0x00007fd093bbb000)
```

I found the following snippet in my log file:
```Updating Effekseer native library at /home/username/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Prominence II RPG- Hasturian Era/minecraft/libEffekseerNativeForJava.so```

I should note this isn't Prism specific, the behavior is the same when using Curseforge and ATLauncher.

In case it helps, this is the truncated crash report:
```
Description: Initializing game

java.lang.RuntimeException: Could not execute entrypoint stage 'client' due to errors, provided by 'aaa_particles'!
	at app//net.fabricmc.loader.impl.FabricLoaderImpl.lambda$invokeEntrypoints$2(FabricLoaderImpl.java:388)
	at app//net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at app//net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:386)
	at app//net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:53)
	at MC//net.minecraft.client.MinecraftClient.<init>(MinecraftClient.java:458)
	at net.minecraft.client.main.Main.main(Main.java:211)
	at app//net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at app//net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at app//net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at app//org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:100)
	at app//org.prismlauncher.EntryPoint.listen(EntryPoint.java:129)
	at app//org.prismlauncher.EntryPoint.main(EntryPoint.java:70)
Caused by: java.lang.ExceptionInInitializerError
	at mod.chloeprime.aaaparticles.client.AAAParticlesClient.installNativeLibrary(AAAParticlesClient.java:55)
	at mod.chloeprime.aaaparticles.client.AAAParticlesClient.init(AAAParticlesClient.java:25)
	at mod.chloeprime.aaaparticles.fabric.AAAParticlesFabricClient.onInitializeClient(AAAParticlesFabricClient.java:15)
	at app//net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:384)
	... 9 more
Caused by: java.lang.UnsatisfiedLinkError: /home/username/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Prominence II RPG- Hasturian Era/minecraft/libEffekseerNativeForJava.so: libGLU.so.1: cannot open shared object file: No such file or directory
	at java.base@17.0.12/jdk.internal.loader.NativeLibraries.load(Native Method)
	at java.base@17.0.12/jdk.internal.loader.NativeLibraries$NativeLibraryImpl.open(NativeLibraries.java:388)
	at java.base@17.0.12/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:232)
	at java.base@17.0.12/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:174)
	at java.base@17.0.12/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2394)
	at java.base@17.0.12/java.lang.Runtime.load0(Runtime.java:755)
	at java.base@17.0.12/java.lang.System.load(System.java:1957)
	at mod.chloeprime.aaaparticles.client.AAAParticlesClient.installNativeLibrary(AAAParticlesClient.java:53)
	... 12 more
```

**Comment by Frontear** [comment]
Hm in your case you're in a flatpak, so you can't access filesystem libs. That being said, it is useful to know that the lib seemingly is replaced at startup.. Either way with the new PR libGLU should no longer be needed.

**Comment by Hello-User** [comment]
> Hm in your case you're in a flatpak, so you can't access filesystem libs. 

No idea why I didn't realize that myself. Using version 1.4.3 on native Prism works.

**Comment by damianethorn** [comment]
Having the same issue but on Mint 21.3 using the Modrinth launcher and the Prominence II modpak, googling the error I found the AAAParticles discussion.

I followed the instructions at https://github.com/ripjackie/EffekseerNativeForJava and just built libEffekseerNativeForJava.so without issue, but am still seeing the same issue (although I am not and never was sure that this is related to a hardcoded dependacy - maybe Im missing something though):

java.lang.RuntimeException: Could not execute entrypoint stage 'client' due to errors, provided by 'aaa_particles'!
	at app//net.fabricmc.loader.impl.FabricLoaderImpl.lambda$invokeEntrypoints$2(FabricLoaderImpl.java:388)
	at app//net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at app//net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:386)
	at app//net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:53)
	at MC//net.minecraft.client.MinecraftClient.<init>(MinecraftClient.java:458)
	at net.minecraft.client.main.Main.main(Main.java:211)
	at app//net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at app//net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at app//net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
Caused by: java.lang.UnsatisfiedLinkError: 'int Effekseer.swig.EffekseerCoreJNI.EffekseerBackendCore_GetDevice()'
	at Effekseer.swig.EffekseerCoreJNI.EffekseerBackendCore_GetDevice(Native Method)
	at Effekseer.swig.EffekseerBackendCore.GetDevice(EffekseerBackendCore.java:56)
	at mod.chloeprime.aaaparticles.api.client.effekseer.Effekseer.getDeviceType(Effekseer.java:24)
	at mod.chloeprime.aaaparticles.client.render.EffekRenderer.init(EffekRenderer.java:39)
	at mod.chloeprime.aaaparticles.fabric.AAAParticlesFabricClient.onInitializeClient(AAAParticlesFabricClient.java:17)
	at app//net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:384)

More details:

    OpenGL version string: 4.6.0 NVIDIA 535.171.04, libopengl 1.4.0-1
	Minecraft Version: 1.20.1
	Minecraft Version ID: 1.20.1
	Operating System: Linux (amd64) version 5.15.0-102-generic
	Java Version: 17.0.12, Azul Systems, Inc.
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode, sharing), Azul Systems, Inc.

ldd libEffekseerNativeForJava.so:

	linux-vdso.so.1 (0x00007ffeb561f000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f7751f3c000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f7751e55000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f7751e35000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f7751c0c000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f7752403000)

**Comment by ripjackie** [comment]
Hmm, It looks like the custom build didn't link against any libOpenGL libs. here's the output of my ldd on `libEffekseerNativeForJava.so`:
```
ldd libEffekseerNativeForJava.so
        linux-vdso.so.1 (0x0000714cf072b000)
        libGLX.so.0 => /usr/lib/libGLX.so.0 (0x0000714cf06cc000)
        libOpenGL.so.0 => /usr/lib/libOpenGL.so.0 (0x0000714cf06a1000)
        libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x0000714cf0000000)
        libm.so.6 => /usr/lib/libm.so.6 (0x0000714cf0311000)
        libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x0000714cf02e3000)
        libc.so.6 => /usr/lib/libc.so.6 (0x0000714cefe0f000)
        /usr/lib64/ld-linux-x86-64.so.2 (0x0000714cf072d000)
        libGLdispatch.so.0 => /usr/lib/libGLdispatch.so.0 (0x0000714cefd57000)
        libX11.so.6 => /usr/lib/libX11.so.6 (0x0000714cefc16000)
        libxcb.so.1 => /usr/lib/libxcb.so.1 (0x0000714cf02b8000)
        libXau.so.6 => /usr/lib/libXau.so.6 (0x0000714cf069a000)
        libXdmcp.so.6 => /usr/lib/libXdmcp.so.6 (0x0000714cf0692000)
```

So it looks like my CMake flags aren't setup right to handle your system? I'll dig into this for you and get back with you.

**Comment by d5c4b3** [comment]
I'm having a similar issue as damianethorn. I compiled Effeseeker and copied over the .so file.

I included a bit more of the crash log where it mentions UnsatisfiedLinkError GLIBC_2.38 not found. 

My ldd is also missing the libOpenGL stuff. I'm on Pop!_OS 22.04


```
java.lang.RuntimeException: Could not execute entrypoint stage 'client' due to errors, provided by 'aaa_particles'!
	at app//net.fabricmc.loader.impl.FabricLoaderImpl.lambda$invokeEntrypoints$2(FabricLoaderImpl.java:388)
	at app//net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at app//net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:386)
	at app//net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:53)
	at MC//net.minecraft.client.MinecraftClient.<init>(MinecraftClient.java:458)
	at net.minecraft.client.main.Main.main(Main.java:211)
	at app//net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at app//net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at app//net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at app//org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:100)
	at app//org.prismlauncher.EntryPoint.listen(EntryPoint.java:129)
	at app//org.prismlauncher.EntryPoint.main(EntryPoint.java:70)
Caused by: java.lang.ExceptionInInitializerError
	at mod.chloeprime.aaaparticles.client.AAAParticlesClient.installNativeLibrary(AAAParticlesClient.java:55)
	at mod.chloeprime.aaaparticles.client.AAAParticlesClient.init(AAAParticlesClient.java:25)
	at mod.chloeprime.aaaparticles.fabric.AAAParticlesFabricClient.onInitializeClient(AAAParticlesFabricClient.java:15)
	at app//net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:384)
	... 9 more
Caused by: java.lang.UnsatisfiedLinkError: /home/cdds/.local/share/PrismLauncher/instances/Prominence II RPG- Hasturian Era/.minecraft/libEffekseerNativeForJava.so: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.38' not found (required by /home/cdds/.local/share/PrismLauncher/instances/Prominence II RPG- Hasturian Era/.minecraft/libEffekseerNativeForJava.so)
	at java.base@17.0.12/jdk.internal.loader.NativeLibraries.load(Native Method)
	at java.base@17.0.12/jdk.internal.loader.NativeLibraries$NativeLibraryImpl.open(NativeLibraries.java:388)
	at java.base@17.0.12/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:232)
	at java.base@17.0.12/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:174)
	at java.base@17.0.12/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2394)
	at java.base@17.0.12/java.lang.Runtime.load0(Runtime.java:755)
	at java.base@17.0.12/java.lang.System.load(System.java:1957)
	at mod.chloeprime.aaaparticles.client.AAAParticlesClient.installNativeLibrary(AAAParticlesClient.java:53)
	... 12 more
```

**Comment by d5c4b3** [comment]
Building directly from the libEffekseer/libEffekseerForMultiLanguages project also produced an `so` file that didn't link against any libOpenGL stuff. 

```
$ ldd build/src/cpp/libEffekseerNativeForJava.so 
    linux-vdso.so.1 (0x00007bc5c66ed000)
    libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007bc5c5c00000)
    libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007bc5c65e3000)
    libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007bc5c65c3000)
    libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007bc5c5800000)
    /lib64/ld-linux-x86-64.so.2 (0x00007bc5c66ef000)
```

**Comment by Frontear** [comment]
Update to this repo, I have successfully managed to compile and link (without GLU) the _original_ effekseer project. I do not believe the original project was to blame for the issues, because my re-compilation works on NVIDIA users again.

See https://github.com/Frontear/effekseer-libs.

I will PR this shortly once I have more confirmation that it works (so far only 2 people have confirmed).
